class7-week3/actions/markAsDone.js

```diff
@@ -0,0 +1,23 @@
+const Todo = require('../models/todo')
+const deserializeTodo = require('../util/deserializeTodo')
+
+module.exports = function markAsDone(request, response) {
+
+  const id = request.params.id
+  const todo = deserializeTodo(request, response)
+  if (todo == null) { return }
+
+  Todo.markAsDone(id, todo.done, (error, todo) => {
```

Comment:
`todo.done` is unnecessary here (but when you pull it out you need to update the function signature in `todo.js` to match).